src/python/pants/backend/shell/target_types.py

```diff
@@ -270,8 +271,41 @@ class ShellCommandOutputsField(StringSequenceField):
     )
 
 
-class ShellCommandDependenciesField(ShellDependenciesField):
+class ShellCommandOutputDependenciesField(ShellDependenciesField):
     supports_transitive_excludes = True
+    alias = "output_dependencies"
+    deprecated_alias = "dependencies"
+    deprecated_alias_removal_version = "2.17.0.dev0"
+
+    help = softwrap(
+        """
+        Any dependencies that the output artifacts require in order to be effectively consumed.
+
+        To enable legacy use cases, if `execution_dependencies` is `None`, these dependencies will
+        be materialized in the command execution sandbox. This behavior is deprecated, and will be
+        removed in version 2.17.0.dev0.
```

Comment:
It's annoying in that `execution_dependencies` will be mandatory-ish for a version or two (if `output_dependencies` is specified), but since `execution_dependencies` feels far more frequent than `output_dependencies` here, it doesn't seem like too big a complaint to hold for a few versions.